### PR TITLE
Feat/optimize featured shelf slider  for accessibility

### PR DESCRIPTION
**Pull request by R-Cloud-Designs**
This Pr aims to improve the accessibility aspects of our Featured Shelf carousel. We wanted to make the user experience smoother by introducing better pagination, ensuring the slides and interactive elements are easily visible, and integrating keyboard navigation in addition to the use of chevron buttons or pagination.

Given the challenges associated with the virtual shelves (carousel) and the technical efforts needed for seamless navigation through pagination, we've recognized that the carousel already maintains a good level of accessibility with keyboard and chevron usage. Therefore, enhancing the navigability of the pagination dots falls into the 'nice-to-have' category, which we might or might not revisit in upcoming tasks.

For now, I've made some adjustments to ensure the pagination dots don't interfere with the accessibility features by setting them to role="presentation" and aria-hidden="true". This way, they won't be unnecessarily announced by screen readers. -- I've dropped a note in the code to clarify our reasoning behind this choice.

Additionally, I've improved the visibility of the chevron's focus state, making it clear when it's selected. 

Link to [Jira Task](https://videodock.atlassian.net/browse/OTT-392)

### CODE REVIEW

File: packages/ui-react/src/components/TileDock/TileDock.tsx [hunk 1]
```diff
@@ -235,8 +235,9 @@ function TileDock<T>({
     if (showDots && isMultiPage && !!renderPaginationDots) {
       const length = pages;
 
+      // Adding role and aria-hidden to the container to avoid screen readers from reading the dots as these is a purely visual indicator
       return (
-        <div className={styles.dots}>
+        <div aria-hidden role="presentation" className={styles.dots}>
```
Comment (langemike) [inline packages/ui-react/src/components/TileDock/TileDock.tsx]: We should apply `aria-hidden="true"`. `aria-hidden` without without an attribute has no purpose and combining it with role="presentation" is weird. It's one or the other.

And the comment should indicate that this is not the ideal solution, but something to improve upon (due to virtualization challenges). Currently the comment states what the `aria-hidden="true"`  / `role="presentation"` purpose is, which has no additional value.

Comment (R-Cloud-Designs) [reply packages/ui-react/src/components/TileDock/TileDock.tsx]: I've removed role presentation in favor of aria-hidden. The comment now reflects de thoughts behind the prop a bit better.
 `// Using aria-hidden="true" due to virtualization issues, making pagination purely visual for now. This is a temporary fix pending a more accessible solution.`

**Comment by langemike** [comment]
Code looks good. But I'm unable to use the chevron buttons to go prev/next. When i swipe (navigate the screenreader) I immediately go to the next element, which is the title of a shelf ("all films"). Happens on voiceover (iOS).

So in short: i cannot go to another slide in the slider.

My Android device acts weird. I'm unable to test this on Talkback for the moment.

Maybe it's my IOS version (15.7.) Have you tested it on the IOS or Android screen reader and what is your experience?

**Comment by langemike** [comment]
> My Android device acts weird. I'm unable to test this on Talkback for the moment.

Suddenly Talkback works. I noticed some issues. I report them in this ticket: https://videodock.atlassian.net/browse/OTT-768

Since this is an improvement over how it was. I approve, lets merge and make it better in the future

**Review by langemike — APPROVED** [review]

**Pull request merged**